### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Andrey1104**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    friends = [Person(someone["name"], someone["age"]) for someone in people]
```
Comment (pavlejviki) [inline app/main.py:12]: choose better variable names.  friends and someone are  not appropriate.

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+    for friend in friends:
```
Comment (pavlejviki) [inline app/main.py:13]: you can simplify the code by first creating all the Person objects, and then, in a subsequent loop, assigning the husbands and wives. At this stage, the husband or wife Person object will have already been created, facilitating this assignment.

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+        for person in people:
+            if "wife" in person \
```
Comment (pavlejviki) [inline app/main.py:15]: use () instead of \

**Review by pavlejviki — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
```
Comment (Dimosphen1) [inline app/main.py:8]: Call `people` not from `self` but from the `Person` class, it is a class attribute.

File: app/main.py [hunk 5]
```diff
@@ -1,8 +1,23 @@
...
+    humans = [Person(person["name"], person["age"]) for person in people]
+    for human in humans:
+        for person in people:
+            if (
+                "wife" in person
+                and person["wife"] is not None
+                and human.name == person["name"]
+            ):
+                human.wife = Person.people[person["wife"]]
+            elif "husband" in person and person["husband"] is not None:
```
Comment (Dimosphen1) [inline app/main.py:21]: You could write this check in one line, e.g.:
```suggestion
            elif person.get("husband"):
```
You could make your previous check shorter as well with the same approach.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

**Pull request closed without merging**